### ISSUE TITLE
Footer is always longer than header with Pagination

### ISSUE THREAD

**Issue by 501warhead** [KyoriPowered/text#39]
This may be a mistake on my end, attempting to integrate this API with my recent works and having some difficulty with pagination. No matter what I attempt to do, the footer is always slightly longer than the header. I've attempted at this point any number of fixes and tried digging for some potential solutions.

I do know this is still a very new feature, so I figured I'd at least provide a bug report for my findings to see if it helps

Version: `3.0.2`

Using `text-serializer-gson`,`text-feature-pagination`, and `text-api` with `text-bukkit-adapter`

Platform is Paper

Width: 53
![image](https://user-images.githubusercontent.com/11152237/61743550-d2360a00-ad95-11e9-857e-5679877e219a.png)

Width: 50
![image](https://user-images.githubusercontent.com/11152237/61743709-217c3a80-ad96-11e9-96d2-c76d44a1d5c0.png)

With longer title, width 50
![image](https://user-images.githubusercontent.com/11152237/61743992-8e8fd000-ad96-11e9-9516-0deab3a9721d.png)

Width 50 with dummy data to cause pagination
![image](https://user-images.githubusercontent.com/11152237/61744659-293cde80-ad98-11e9-873b-e89252f969f0.png)

smol title
![image](https://user-images.githubusercontent.com/11152237/61746801-c863d500-ad9c-11e9-83e5-3d036ea04e7c.png)

Code (In Kotlin) ((An abomination of formatting, I apologize in advance))
```Kotlin
return Pagination.builder().line { cs ->
            cs.character('=')
        }.width(53).build(TextComponent.of("Charters").color(TextColor.GOLD), Pagination.Renderer.RowRenderer { value, index ->
            return@RowRenderer listOf(if (value == null) TextComponent.of("${index + 1}. ").color(TextColor.YELLOW).content("ERR?").color(TextColor.RED) else
                TextComponent.of("${index + 1}. ")
                        .color(TextColor.YELLOW)
                        .append(TextComponent.of(value.name).color(TextColor.GRAY))
                        .hoverEvent(HoverEvent.of(HoverEvent.Action.SHOW_TEXT, TextComponent.of("Click to view this charter").color(TextColor.GRAY)))
                        .clickEvent(ClickEvent.of(ClickEvent.Action.RUN_COMMAND, "/charter info ${value.name}")))
        }, Pagination.PageCommandFunction { page -> "/charter list -page $page" })
```

```Kotlin
        val rendered = settlementService.buildSettlementPagination().render(settlementService.dummyData().sortedBy { settlement -> settlement.name }, page)
        rendered.forEach { comp ->
            TextAdapter.sendComponent(p, comp)
        }
```

**Comment by lucko** [comment]
If you test in the console, they will match up ;)

This is because characters have different pixel lengths in the default Minecraft font.

**Comment by 501warhead** [comment]
That makes sense, but that's a bit difficult isn't it? Doesn't that mean if you use any non-minecraft language selection (I forget what they're called currently, Unicode fonts maybe?) it'd break if you automatically adjusted for the Minecraft font?

**Comment by lucko** [comment]
Yes, we could adjust for the Minecraft chat pixel lengths - but then it would be wrong in console, or if you used a different font in your resource pack.

**Comment by 501warhead** [comment]
Alright. I think I have a few hacky solutions to try then now that I understand the problem a bit better, though maybe it'd be nice to have some ability to disable the extra character padding? I don't think I saw anything obvious when I was using the API, though maybe setting style character to null might be it?

Anyway, thank you for the information!
